File: Backend/src/main/java/com/workout/tracker/entity/Exercise.java
package com.workout.tracker.entity;

import com.workout.tracker.model.MuscleGroup;
import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
public class Exercise {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private Integer sets;

    private Integer reps;
    private Double weight;
    private Integer rir;
    private boolean completed;

    @Enumerated(EnumType.STRING)
    private MuscleGroup muscleGroup;

    @ManyToOne
    @JoinColumn(name = "workout_per_day_id", nullable = false)
    private WorkoutPerDay workoutPerDay;

}

File: Frontend/src/app/class/exercise.ts
export enum MuscleGroup {
  BACK = 'BACK',
  CHEST = 'CHEST',
  SHOULDERS = 'SHOULDERS',
  BICEPS = 'BICEPS',
  TRICEPS = 'TRICEPS',
  LEGS = 'LEGS',
  CORE = 'CORE',
  CARDIO = 'CARDIO',
}

export class Exercise {
  public name: string;

  public sets: number;

  public reps: number;
  public weight: number;
  public rir: number;
  public completed: boolean;

  public muscleGroup: MuscleGroup;

  public constructor(
    name: string,
    muscleGroup = MuscleGroup.BACK,
    sets = 0,
    reps = 0,
    weight = 0,
    rir = 0,
    completed = false
  ) {
    this.name = name;
    this.muscleGroup = muscleGroup;
    this.sets = sets;
    this.reps = reps;
    this.weight = weight;
    this.rir = rir;
    this.completed = completed;
  }
}
